admin/admin-default-term.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * Manages filters and actions related to default terms.
+ *
+ * @since 3.0
+ */
+class PLL_Admin_Default_Term {
+
+	/**
+	 * A reference to the PLL_Model instance.
+	 *
+	 * @since 2.8
+	 *
+	 * @var PLL_Model
+	 */
+	protected $model;
+
+	/**
+	 * Preferred language to assign to new contents.
+	 *
+	 * @var PLL_Language
+	 */
+	protected $pref_lang;
+
+	/**
+	 * Reference to Polylang options array
+	 *
+	 * @var array $options
+	 */
+	protected $options;
+
+	/**
+	 * @param object $polylang
+	 */
+	public function __construct( &$polylang ) {
+		$this->model = &$polylang->model;
+		$this->pref_lang = &$polylang->pref_lang;
+		$this->options = &$polylang->options;
+	}
+
+	/**
+	 * Setups filters and actions needed.
+	 *
+	 * @return void
+	 */
+	public function add_hooks() {
+		$taxonomies = get_taxonomies();
+		foreach ( $taxonomies as $taxonomy ) {
+			if ( 'category' === $taxonomy ) {
+				// Allows to get the default categories in all languages
+				add_filter( 'option_default' . $taxonomy, array( $this, 'option_default_category' ) );
+				add_action( 'update_option_default_' . $taxonomy, array( $this, 'update_option_default_category' ), 10, 2 );
+				add_action( 'pll_add_language', array( $this, 'handle_default_category_on_create_language' ) );
+
+				// The default category should be in the default language
+				add_action( 'pll_update_default_lang', array( $this, 'update_default_category_language' ) );
+
+				// Adds the language column in the 'Categories' table.
+				add_filter( 'manage_' . $taxonomy . '_custom_column', array( $this, 'term_column' ), 10, 3 );
+			}
+
+			// Adds the language column in the 'Post Tags'table.
+			if ( 'post_tag' === $taxonomy ) {
+				add_filter( 'manage_' . $taxonomy . '_custom_column', array( $this, 'term_column' ), 10, 3 );
+			}
+		}
+
+		// Prevents deleting all the translations of the default category
+		add_filter( 'map_meta_cap', array( $this, 'fix_delete_default_category' ), 10, 4 );
+	}
+
+	/**
+	 * Filters the default category in note below the category list table and in settings->writing dropdown
+	 *
+	 * @since 1.2
+	 *
+	 * @param int $value
```

Comment:
Ad short descriptions to all the function parameters.

Reply:
What about we don't force ourselves on documenting everything even we have no good information to add?

Let's imagine we rewrite this one like the following:

```php
 * @param int $value An integer value.
```

What would be the benefits?

Reply:
> What about we don't force ourselves on documenting everything even we have no good information to add?
> 
> Let's imagine we rewrite this one like the following:
> 
> ```
>  * @param int $value An integer value.
> ```
> 
> What would be the benefits?

Obviously said like that it doesn't make sense 😁
However as the `option_{$option}` filter could pass a $value `mixed` variable
See https://developer.wordpress.org/reference/hooks/option_option/
 it could be interesting to precise in the case of taxonomy that this integer value corresponds on the taxonomy term id for the option `default_{$taxonomy}`

Reply:
Indeed, the fact that the parameter is a taxonomy term id is not obvious in this case. So what about renaming the variable to a more fitting name like `$taxonomy_term_id` ?

Reply:
It could 👍 What is the opinion of each other ? 
Preferable to document by naming variable or by commenting or both 🙂 ?

Reply:
We might as well do both.